diff --git a/pbs-api-types/src/crypto.rs b/pbs-api-types/src/crypto.rs
index xxxxxxxx..yyyyyyyy 100644
--- a/pbs-api-types/src/crypto.rs
+++ b/pbs-api-types/src/crypto.rs
@@ -88,7 +88,7 @@ pub mod bytes_as_fingerprint {
         let mut s = String::deserialize(deserializer)?;
         s.retain(|c| c != ':');
         let mut out = MaybeUninit::<[u8; 32]>::uninit();
-        hex::decode_to_slice(s.as_bytes(), unsafe { &mut (*out.as_mut_ptr())[..] })
+        hex::decode_to_slice(s.as_bytes(), unsafe { &mut (&mut (*out.as_mut_ptr()))[..] })
             .map_err(serde::de::Error::custom)?;
         Ok(unsafe { out.assume_init() })
     }
