diff --git a/pbs-api-types/src/crypto.rs b/pbs-api-types/src/crypto.rs
index xxxxxxxx..yyyyyyyy 100644
--- a/pbs-api-types/src/crypto.rs
+++ b/pbs-api-types/src/crypto.rs
@@ -88,7 +88,7 @@ impl std::str::FromStr for Fingerprint {
         if s.len() != 64 {
             bail!("bad fingerprint length");
         }
-        hex::decode_to_slice(s.as_bytes(), unsafe { &mut (*out.as_mut_ptr())[..] })
+        hex::decode_to_slice(s.as_bytes(), unsafe { &mut (&mut (*out.as_mut_ptr()))[..] })
             .map_err(|_| format_err!("bad fingerprint"))?;
         Ok(Self(out))
     }
