diff --git a/proxmox-uuid/src/lib.rs b/proxmox-uuid/src/lib.rs
index xxxxxxxx..yyyyyyyy 100644
--- a/proxmox-uuid/src/lib.rs
+++ b/proxmox-uuid/src/lib.rs
@@ -97,7 +97,7 @@ impl Uuid {
     pub fn generate() -> Self {
         let mut uuid = Self::null();
         unsafe {
-            let uuid: &mut [u8] = unsafe { &mut (*uuid)[..] };
+            let uuid: &mut [u8] = unsafe { &mut (&mut (*uuid))[..] };
             if libc::getrandom(
                 uuid.as_mut_ptr() as *mut libc::c_void,
                 uuid.len(),
@@ -118,7 +118,7 @@ impl Uuid {
     pub fn generate_time_based() -> Self {
         let mut uuid = Self::null();
         unsafe {
-            let uuid: &mut [u8] = unsafe { &mut (*uuid)[..] };
+            let uuid: &mut [u8] = unsafe { &mut (&mut (*uuid))[..] };
             if libc::getrandom(
                 uuid.as_mut_ptr() as *mut libc::c_void,
                 uuid.len(),
